File: lab-oop-1/C++/Triangle.cpp
#include "Triangle.h"
#include <cmath>

Triangle::Triangle(double _x1, double _y1, double _x2, double _y2, double _x3, double _y3) {
    x1 = _x1; y1 = _y1;
    x2 = _x2; y2 = _y2;
    x3 = _x3; y3 = _y3;
}

double Triangle::getPerimeter() {

    double a = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
    double b = sqrt(pow(x3 - x2, 2) + pow(y3 - y2, 2));
    double c = sqrt(pow(x1 - x3, 2) + pow(y1 - y3, 2));
    return a + b + c;
}

double Triangle::getArea() {

    return abs((x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) / 2.0);
}

File: lab-oop-1/C#/Triangle.cs
using System;

namespace lab_1
{
    public class Triangle
    {
       
        private double x1, y1, x2, y2, x3, y3;

       
        public Triangle(double x1, double y1, double x2, double y2, double x3, double y3)
        {
            this.x1 = x1; this.y1 = y1;
            this.x2 = x2; this.y2 = y2;
            this.x3 = x3; this.y3 = y3;
        }
        
        public double GetPerimeter()
        {
            double a = Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
            double b = Math.Sqrt(Math.Pow(x3 - x2, 2) + Math.Pow(y3 - y2, 2));
            double c = Math.Sqrt(Math.Pow(x1 - x3, 2) + Math.Pow(y1 - y3, 2));
            return a + b + c;
        }

      
        public double GetArea()
        {
            return Math.Abs((x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) / 2.0);
        }
    }
}